resource/client.go

```diff
@@ -158,6 +161,20 @@ type WatchEvent struct {
 	Object Object
 }
 
+// CustomRouteRequestOptions contains the options for a custom route request
+type CustomRouteRequestOptions struct {
+	// Path is the path of the custom route, from the base of the resource or namespace/api version
+	Path string
```

Comment:
Do you think we should make an example here, to better explain how a path should be?